server/config/validate-config-keys.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+const chalk = require( 'chalk' );
+const fs = require( 'fs' );
+const path = require( 'path' );
+
+/** @type {String} path to configuration file directory */
+const configRoot = path.resolve( __dirname, '../../', 'config' );
+
+/**
+ * @type {RegExp} matches possible config files
+ *
+ *  - doesn't start with an underscore
+ *  - ends in `.json`
+ */
+const environmentConfigPattern = /^(?!_)[^.]+\.json$/;
+
+/**
+ * Attempts to parse JSON data
+ *
+ * If data doesn't parse it will return
+ * `null` instead of throwing an error
+ *
+ * @param {*} data
+ * @returns {*}
+ */
+const parseJSON = data => {
+	try {
+		return JSON.parse( data );
+	} catch (e) {
+		return null;
+	}
+};
+
+/** @type {String[]} list of configuration files */
+const environmentConfigFiles = fs
+	.readdirSync( configRoot, { encoding: 'utf8' } )
+	.filter( filename => environmentConfigPattern.test( path.basename( filename ) ) )
+	.filter( filename => ! ( /secrets/g ).test( filename ) )
+	.filter( filename => ! ( /client/g ).test( filename ) );
```

Comment:
written inefficiently but I was trying to make a straight-forward list of what _will_ and what _won't_ be considered an environment-specific config file. as I don't understand this system too well I would appreciate some confirmation or education here